adapter/internal/discovery/xds/server.go

```diff
@@ -301,17 +301,32 @@ func UpdateAPI(vHost string, apiProject model.ProjectAPI, environments []string)
 		// it will enable folowing securities globally for the API, overriding swagger securities.
 		isYamlAPIKey := false
 		isYamlOauth := false
+		isYamlMutualssl := false
+		isYamlMutualsslMandatory := false
+		isYamlOauthBasicAuthAPIKeyMandatory := false
 		for _, value := range apiYaml.SecurityScheme {
 			if value == constants.APIMAPIKeyType {
 				logger.LoggerXds.Debugf("API key is enabled in api.yaml for API %v:%v", apiYaml.Name, apiYaml.Version)
 				isYamlAPIKey = true
 			} else if value == constants.APIMOauth2Type {
 				logger.LoggerXds.Debugf("Oauth2 is enabled in api.yaml for API %v:%v", apiYaml.Name, apiYaml.Version)
 				isYamlOauth = true
+			} else if value == constants.APIMMutualSSLType {
+				logger.LoggerXds.Debugf("Mutual SSL is enabled in api.yaml for API %v:%v", apiYaml.Name, apiYaml.Version)
+				isYamlMutualssl = true
+			} else if value == constants.APIMMutualSSLMandatoryType {
+				logger.LoggerXds.Debugf("Mutual SSL Mandatory is enabled in api.yaml for API %v:%v", apiYaml.Name, apiYaml.Version)
+				isYamlMutualsslMandatory = true
+			} else if value == constants.APIOauthBasicAuthAPIKeyMandatoryType {
+				isYamlOauthBasicAuthAPIKeyMandatory = true
 			}
 		}
-		mgwSwagger.SanitizeAPISecurity(isYamlAPIKey, isYamlOauth)
+		mgwSwagger.SanitizeAPISecurity(isYamlAPIKey, isYamlOauth, isYamlMutualssl, isYamlMutualsslMandatory, isYamlOauthBasicAuthAPIKeyMandatory)
 		mgwSwagger.SetOperationPolicies(apiProject)
+		mgwSwagger.SetXWso2AuthHeader(apiYaml.AuthorizationHeader)
+	} else if apiYaml.APIType != constants.WS {
```

Comment:
Do we need this?

Reply:
No, we do not need that. It is a mistake when merging. I will remove it.

Reply:
Removed